src/storage/compactor/src/server.rs

```diff
@@ -57,11 +57,13 @@ pub async fn compactor_serve(
 
     // Register to the cluster.
     let meta_client =
-        MetaClient::register_new(&opts.meta_address, WorkerType::Compactor, &client_addr, 0)
+        MetaClient::register_new(&opts.meta_address, WorkerType::ComputeNode, &client_addr, 0)
```

Comment:
？？

Reply:
I am using `WorkerType::Unspecified` now. Is that better?

Reply:
This's not a test but a regular register logic for compactor. You shouldn't suppose to change it. Anyway, I just found it randomly.

Reply:
If I am using `WorkerType::Unspecified` I get `gRPC error (Internal error): worker_type`. For all other types I get `gRPC error (Operation is not implemented or not supported)`

Reply:
Don't use `WorkerType::Unspecified`, use `WorkerType::ComputeNode` or `WorkerType::Frontend` instead.

`gRPC error (Operation is not implemented or not supported)` means the server you required don't enable target service, which could be possible the follower node. If using `register_new` to connect to meta nodes, it already includes a RPC call `add_worker_node` to cluster service that only enabled in leader node. So you don't have to call `send_heartbeat` to to determine whether it is a leader or a follower, you can tell if it's  a leader or follower simply by whether the return value is ok or err of `register_new`.

Reply:
`GrpcMetaClient` is not public, or we can just create a cluster client via
```
let cluster_client = ClusterServiceClient::new(channel.clone());
```
like what we did in `GrpcMetaClient:new(..)`. After meta cluster member service get implemented, we can replace it in the unit tests.